source/shared/cpp/ObjectModel/Enums.cpp

```diff
@@ -1101,6 +1126,30 @@ namespace AdaptiveSharedNamespace
         return verticalContentAlignmentNameToEnum[verticalContentAlignment];
     }
 
+    const std::string SentimentToString(Sentiment sentiment)
+    {
+        std::unordered_map<Sentiment, std::string, EnumHash> sentimentEnumToName;
+        GetSentimentEnumMappings(&sentimentEnumToName, nullptr);
+
+        if (sentimentEnumToName.find(sentiment) == sentimentEnumToName.end())
+        {
+            throw std::out_of_range("Invalid VerticalContentAlignment");
```

Comment:
wrong error message

Reply:
it appears that we have many duplicate of the same code here. perhpas we can refactor it.

---
In reply to: [226447317](https://github.com/Microsoft/AdaptiveCards/pull/2036#discussion_r226447317) [](ancestors = 226447317)